### PR TITLE
Set up vitest workspaces, re-enable ssr test

### PR DESCRIPTION
**Pull request by nstepien**
https://vitest.dev/guide/workspace.html

### CODE REVIEW

File: test/node/ssr.test.tsx [hunk 1]
```diff
@@ -23,14 +22,10 @@ function App() {
...
-test.skip('basic server-side rendering (SSR) support', () => {
+test('basic server-side rendering (SSR) support', () => {
   // make sure we're not running in the JSDOM environment
   expect(globalThis.window).toBeUndefined();
 
   const html = renderToString(<App />);
   expect(html).not.toHaveLength(0);
-
-  // make sure `@testing-library/jest-dom` does not polyfill `window.CSS`
-  expect(typeof CSS).toBe('undefined');
```
Comment (nstepien) [inline test/node/ssr.test.tsx:35]: https://github.com/adazzle/react-data-grid/commit/b9e3344cd53c96b45a4fa3dbb6c5d0a9425f690e
Seems like this hasn't been needed for a while.

File: vite.config.ts [hunk 2]
```diff
@@ -56,14 +41,7 @@ export default defineConfig(({ command }) => ({
...
     setupFiles: ['test/setup.ts'],
-    browser: {
-      enabled: true,
-      name: 'chromium',
-      provider: 'playwright',
-      commands: { resizeColumn },
-      viewport: { width: 1920, height: 1080 },
-      headless: true
-    },
+    reporters: ['basic'],
```
Comment (nstepien) [inline vite.config.ts:44]: The default reports just spams the console, and we missed a console log because of it.

File: test/node/ssr.test.tsx [hunk 3]
```diff
@@ -23,14 +22,10 @@ function App() {
   return <DataGrid columns={columns} rows={rows} />;
 }
 
-// biome-ignore lint/suspicious/noSkippedTests: <explanation>
-test.skip('basic server-side rendering (SSR) support', () => {
+test('basic server-side rendering (SSR) support', () => {
   // make sure we're not running in the JSDOM environment
```
Comment (amanmahajan7) [inline test/node/ssr.test.tsx:26]:
```suggestion
```
We do not have JSDOM package anymore. Can we remove this comment?

Comment (nstepien) [reply test/node/ssr.test.tsx]: Tweaked it

**Review by amanmahajan7 — APPROVED** [review]

**Pull request merged**